components/block/element.js

```diff
@@ -9,61 +10,119 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    loading: {type: Boolean, state: true},
+    blockId: {type: String},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
+  constructor() {
+    super();
+
+    this.loading = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.model.on('sidebar_save:success', this.refresh.bind(this));
+  }
+
+  get layout() {
+    return window.parent.Core.g.layout;
+  }
+
+  get model() {
+    return (this.cached_model ||= this.layout.blocks.findWhere({
+      id: this.blockId,
+    }));
+  }
+
+  get parentModel() {
+    if (this.cached_parent_model !== undefined) return this.cached_parent_model;
+
+    const parentId = this.model.attributes.parent_block_id;
+    const parentModel = document.querySelector(
+      `ngl-block[blockId="${parentId}"]`
+    );
```

Comment:
extract this into dedicated getter

Reply:
I am not quite sure what you mean here -- A separate, non-caching getter for the parent block element?

Reply:
With all new changes I've added a different suggestion.